homeassistant/components/kat_bulgaria/binary_sensor.py

```diff
@@ -0,0 +1,96 @@
+"""Setup for the binary sensor from the configuration.yaml."""
+
+from datetime import datetime, timedelta
+import logging
+
+from kat_bulgaria.obligations import (
+    REGEX_DRIVING_LICENSE,
+    REGEX_EGN,
+    KatError,
+    KatFatalError,
+    KatPersonDetails,
+    check_obligations,
+)
+import voluptuous as vol
+
+from homeassistant.components.binary_sensor import PLATFORM_SCHEMA, BinarySensorEntity
+from homeassistant.core import HomeAssistant
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+
+from .const import (
+    BINARY_SENSOR_ENTITY_PREFIX,
+    BINARY_SENSOR_NAME_PREFIX,
+    CONF_DRIVING_LICENSE,
+    CONF_PERSON_EGN,
+    CONF_PERSON_NAME,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = "kat_bulgaria"
+
+SCAN_INTERVAL = timedelta(minutes=20)
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_PERSON_EGN): vol.All(cv.string, vol.Match(REGEX_EGN)),
```

Comment:
We no longer accept additions or changes to the yaml config of integrations. See: https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md

This integration needs to be refactored to use a config flow and config entries.

https://developers.home-assistant.io/docs/config_entries_config_flow_handler